File: src/com/company/App/AppCommands/Redo.java
#pragma once

#include "BaseCommand.h"

namespace Apps {
	/*
	|--------------------------------------------------------------------------
	| Redo is a class for Redo command. Redo call the AppRedo to load memory.
	*/
	class App::Redo : public BaseCommand {
	public:
		/*********************************************************************************
		* Default constructor
		**********************************************************************************
		*/
		Redo();
		/*********************************************************************************
		* Default constructor
		**********************************************************************************
		*/
		Redo(string);
		/*********************************************************************************
		* Default destructor
		**********************************************************************************
		*/
		~Redo();
		/*********************************************************************************
		* Execute command
		**********************************************************************************
		*/
		void Exec(App&) const;
		/*********************************************************************************
		* Create new instance of the class
		**********************************************************************************
		*/
		static ICommand* Create(string);
		/*********************************************************************************
		* Check if string input is appropriate for this class
		**********************************************************************************
		*/
		static bool Validate(string);
	private:
		int n;
	};
}

File: src/com/company/App/src/Redo.cpp
#include "../stdafx.h"
#include "../AppCommands/Redo.h"

namespace Apps {
	App::Redo::Redo() {
		m_commandName = "Redo";
	}
	App::Redo::Redo(string input) {
		m_commandName = "Redo";
		m_input = input;
		istringstream tmp(input);
		string kata1;
		tmp >> kata1;
		if (!(tmp >> n))
			n = 1;
	}
	App::Redo::~Redo() {}
	void App::Redo::Exec(App& app) const {
		app.AppRedo(n);
	}
	ICommand* App::Redo::Create(string input) {
		return new Redo(input);
	}
	bool App::Redo::Validate(string input) {
		/* Get the symbols from the input */
		if (input.substr(0, 4) == "Redo") return true;
		else return false;
	}
}